lib/engine/connection.rb

```diff
@@ -5,39 +5,25 @@ class Connection
     attr_reader :paths
 
     def self.connect!(hex)
-      connections = {}
-      node_paths = []
-      hex_edges = {}
-
       hex.tile.paths.each do |path|
-        path.walk { |p| node_paths << p if p.node? }
-      end
-
-      node_paths.uniq.each do |node_path|
-        node_path.walk(chain: []) do |chain|
+        path.walk(chain: []) do |chain|
           next unless valid_connection?(chain)
 
+          path = chain[0]
```

Comment:
this still preserves the old code? it looks difference because you're no longer using chains?

Reply:
The old code did both hexes in a chain, this just does the one it works on